packages/elements/src/components/ino-carousel/ino-carousel.tsx

```diff
@@ -0,0 +1,168 @@
+
+  /**
+   * Hides the arrow buttons
+   */
+  @Prop() inoHideButtons: boolean = false;
+
+  /**
+   * Restarts playback from the first slide upon reaching the last slide
+   */
+  @Prop() inoInfinite: boolean = false;
+
+  /**
+   * Sets the intermission between two slides (Unit: ms)
+   */
+  @Prop() inoIntermission: number = 5000;
+
+  /**
+   * Enables reverse playback of the slides
+   */
+  @Prop() inoReverse: boolean = false;
+
+  componentDidLoad(): void {
+    this.slides = this.getSlides();
+    this.configureSlides();
+    this.configureAutoplay();
+  }
+
+  /**
+   * adds a slide animation to the slide with the given index
+   * required to prevent the animation from playing when loading the component
+   * @param slide carousel slide
+   */
+  private addAnimationToSlide = (slide: HTMLInoCarouselSlideElement) => {
+    if (!this.inoAnimated) return;
+    if (!slide.classList.contains('ino-carousel--animated')) {
+      slide.classList.add('ino-carousel--animated');
+    }
+  };
+
+  private configureSlides = () => {
+    if (this.slides.length < 1) return;
+    if (!this.selectSlide(this.value)) this.slides[this.currentSlide].classList.add('ino-carousel-slide--selected');
+  };
+
+  private selectSlide(value: number): boolean {
+    let slideSelected = false;
+    this.slides.forEach((slide) => {
+      if (value === slide.value) {
+        slide.classList.add('ino-carousel-slide--selected');
+        this.currentSlide = this.slides.indexOf(slide);
+        slideSelected = true;
+      } else {
+        slide.classList.remove('ino-carousel-slide--selected');
+      }
+    });
+    return slideSelected;
+  }
+
+  private configureAutoplay = () => {
+    if (this.slides.length < 1) return;
+    if (this.inoAutoplay) {
+      this.timer = setInterval(this.nextSlide, this.inoIntermission);
+      return;
+    }
+    clearInterval(this.timer);
+  };
+
+  /**
+   * Switches to the next slide
+   * Required for autoplay to work
+   */
+  private nextSlide = () => {
+    if (this.slides.length < 1) throw new Error('There are no slides to display');
+    this.slides[this.currentSlide].classList.remove('ino-carousel-slide--selected');
+    this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to current slide
+    this.getNextSlide();
+    this.addAnimationToSlide(this.slides[this.currentSlide]); // adds the slide animation to the new slide
+    this.slides[this.currentSlide].classList.add('ino-carousel-slide--selected');
+    this.slideCounter++;
+
+    // disables the timer after all slides have been shown
+    if (!this.inoInfinite && this.slideCounter >= this.slides.length) {
+      this.slideCounter = 1;
+      clearInterval(this.timer);
+    }
+  };
+
+  private getSlides() {
+    return Array.from(this.el.querySelectorAll('ino-carousel-slide')) as HTMLInoCarouselSlideElement[];
+  };
+
+  private mod = (a, b) => ((a % b) + b) % b;
+
+  /**
+   * determines the index of the next slide
```

Comment:
This comment is outdated as well

Reply:
Actually, that's exactly what it does. The `getNextSlide` function determines and returns the index of the next slide and not the slide itself.

Reply:
Well yea, now it does 😄